### PR TITLE
[CONFORMANCE][TOOLS] New SubgraphsDumper

### PR DESCRIPTION
**Pull request by iefode**
### Details:
 - *Cache was separated to Interface class, op_cache, graph_cache*
 - *Matchers were used for single op matching*
 - *Meta info was implemented as single structure*
 - *All constants are being stored as parameters with ranges instead of constant values*
 - *Tests were extended*

### Tickets:
 - *109953 - Reduce graph quantity*
 - *109963 - test coverage*
 - *109957 - Graph comparation*

subgraphs_dumper

### CODE REVIEW

File: src/tests/functional/plugin/conformance/subgraphs_dumper/include/utils/model_utils.hpp [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+#endif
+#ifdef ENABLE_OV_IR_FRONTEND
+    result.push_back(std::regex(R"(.*\.xml)"));
+#endif
+#ifdef ENABLE_OV_TF_LITE_FRONTEND
+    result.push_back(std::regex(R"(.*\.tflite)"));
+#endif
+#ifdef ENABLE_OV_PYTORCH_FRONTEND
+    result.push_back(std::regex(R"(.*\.pt)"));
+#endif
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/include/utils/model_utils.hpp:37]: why do we need it?

File: src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/op_cache.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+// protected:
+//     SerializationStatus serialize_op(const std::pair<std::shared_ptr<ov::Node>, LayerTestsUtils::GraphMetaInfo> &op_info);
+
+//     std::map<std::shared_ptr<ov::Node>, LayerTestsUtils::GraphMetaInfo> m_ops_cache;
+// }
+
+// }  // namespace subgraph_dumper
+// }  // namespace tools
+// }  // namespace ov
```
Comment (ilya-lavrenov) [inline src/tests/functional/plugin/conformance/subgraphs_dumper/include/cache/op_cache.cpp:29]: remove comment code

File: src/common/util/include/openvino/util/file_util.hpp [hunk 3]
```diff
@@ -163,6 +163,28 @@ inline int64_t file_size(const char* path) {
     return in.tellg();
 }
 
+/**
```
Comment (iefode) [inline src/common/util/include/openvino/util/file_util.hpp:166]: @ilya-lavrenov Could you please take a look of `file_util.hpp`?

**Review by ilya-lavrenov — COMMENTED** [review]
reviewed only common part

File: src/common/util/include/openvino/util/file_util.hpp [hunk 4]
```diff
@@ -193,7 +233,7 @@ inline int64_t file_size(const std::string& path) {
 template <typename C,
           typename = typename std::enable_if<(std::is_same<C, char>::value || std::is_same<C, wchar_t>::value)>::type>
 inline bool file_exists(const std::basic_string<C>& path) {
-    return file_size(path) > 0;
+    return file_exists(path);
 }
```
[the remark this answers is not in the document]
Comment (iefode) [reply src/common/util/include/openvino/util/file_util.hpp:237]: Do u mean `file_size`?

[the remark this answers is not in the document]
Comment (ilya-lavrenov) [reply src/common/util/include/openvino/util/file_util.hpp:237]: `inline bool file_exists(const std::basic_string<C>& path)  `

[the remark this answers is not in the document]
Comment (iefode) [reply src/common/util/include/openvino/util/file_util.hpp:237]: @moved to separated PR: https://github.com/openvinotoolkit/openvino/pull/18187/files with fixed comments. Thanks!

**Review by sbalandi — APPROVED** [review]

**Pull request merged**